### PR TITLE
sys/console: fix console compatibility

### PR DESCRIPTION
**Pull request by michal-narajowski**
The previous fix (#408, MYNEWT-811) introduced a regression in handling CR/LF characters.
When CR/LF was used, the LF character was added to the buffer and handled by the shell resulting in "Unrecognized command" error.
Additionally, this patch also fixes these issues in console/minimal.

### CODE REVIEW

File: sys/console/minimal/src/console.c [hunk 1]
```diff
@@ -86,11 +86,17 @@ console_read(char *str, int cnt, int *newline)
...
     len = strlen(cmd->line);
+
     if ((cnt - 1) < len) {
         len = cnt - 1;
+        if (len > 0) {
+            memcpy(str, cmd->line, len);
+            str[len] = '\0';
+        } else {
+            str[len] = cmd->line[0];
+        }
```
Comment (mkiiskila) [inline sys/console/minimal/src/console.c]: This is broken. The common case, where cnt > len, would copy no bytes to str.

Also, I don't understand line 96 here. Why are you setting last character in str to be
first character of input?

Comment (vrahane) [inline sys/console/minimal/src/console.c]: @michal-narajowski, I agree with @mkiiskila.  Can you remove changes to this file for the time being as it is going to break boot-serial. I will fix it in console/full and console/minimal and make sure it works with boot-serial as well. I will open a new PR with a fix to both.

File: sys/console/minimal/src/console.c [hunk 2]
```diff
@@ -222,6 +229,13 @@ console_handle_char(uint8_t byte)
             nlip_state |= NLIP_DATA_START1;
             break;
         case '\r':
```
Comment (vrahane) [inline sys/console/minimal/src/console.c]: /* Falls through. */ comment is needed here as GCC7 gives an error without an explicit commented fallthrough.

File: sys/console/full/src/console.c [hunk 3]
```diff
@@ -473,10 +474,14 @@ console_handle_char(uint8_t byte)
         case ESC:
             esc_state |= ESC_ESC;
             break;
-        default:
-            insert_char(&input->line[cur], byte, end);
-            /* Falls through. */
         case '\r':
```
Comment (vrahane) [inline sys/console/full/src/console.c]: /* Falls through. */ comment is needed here as GCC7 gives an error without an explicit commented fallthrough.

**Comment by michal-narajowski** [comment]
Updated.

**Pull request merged**